src/service/referral-api/src/FamilyHubs.Referral.Core/Queries/GetReferrals/GetReferralCountByServiceIdCommand.cs

```diff
@@ -0,0 +1,34 @@
+using FamilyHubs.Referral.Data.Repository;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace FamilyHubs.Referral.Core.Queries.GetReferrals;
+
+public class GetReferralCountByServiceIdCommand : IRequest<int>
+{
+    public long ServiceId { get; }
+
+    public GetReferralCountByServiceIdCommand(long serviceId)
+    {
+        ServiceId = serviceId;
+    }
+}
+
+public class GetReferralCountByServiceIdCommandHandler : IRequestHandler<GetReferralCountByServiceIdCommand, int>
+{
+    private readonly ApplicationDbContext _context;
+
+    public GetReferralCountByServiceIdCommandHandler(ApplicationDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<int> Handle(GetReferralCountByServiceIdCommand request, CancellationToken cancellationToken)
+    {
+        IQueryable<Data.Entities.Referral> query = _context.Referrals
+            .AsNoTracking()
+            .Where(r => r.ReferralServiceId == request.ServiceId);
```

Comment:
Are Accepted/Declined referrals filtered?

Reply:
No they are not, good spot..

Is it just literally "Opened" that counts or do "New", "Accepted" and "Declined" also count (in which case that query is enough, but dunno)

Reply:
I think they get created as "New".
They transition to "Opened" if a Vcs user looks at them and clicks "Return later".
"Accepted" and "Declined" are final states.

imo both "New" and "Opened" should count but I don't know

Reply:
Filtering by New & Opened: [128e76f](https://github.com/DFE-Digital/fh-services/pull/121/commits/128e76fb1883d8f6c82bbf1e36dd7cc32e3a852e)